### DISCUSSION TITLE
Alerting action trigger script

### DISCUSSION THREAD

**Discussion by wooyoungAhn** [NetApp/harvest#2593] [Q&A] [answered]
Hello everyone
I am studying to trigger a script when prometheus Alerting actions like Active IQ Unified Manager.
If anyone has already implemented this, can you advise me?

https://docs.netapp.com/us-en/active-iq-unified-manager-99/online-help/concept-how-scripts-work-with-alerts.html

**Comment by rahulguptajss** [comment, marked as answer]
@wooyoungAhn You can use prometheus alert manager through a webhook. Below are the sample steps. 

## Setting Up a Webhook Receiver with Golang and Prometheus Alertmanager

### Prerequisites

- Prometheus and Alertmanager installed and configured.
- Golang installed

### Step 1: Create the Go Webhook Receiver

Create a new Go file (e.g., `main.go`) and add the following code. You can invoke a script in go code as needed.

```go
package main

import (
	"fmt"
	"io"
	"log"
	"net/http"
)

func alertHandler(w http.ResponseWriter, r *http.Request) {
	body, err := io.ReadAll(r.Body)
	if err != nil {
		http.Error(w, "Error reading request body",
			http.StatusInternalServerError)
		return
	}

	// Print the received data
	fmt.Println(string(body))
}

func main() {
	http.HandleFunc("/", alertHandler)
	log.Fatal(http.ListenAndServe(":5001", nil))
}
```

This code creates a simple web server that listens for POST requests at the `/` endpoint and prints the received alert data.

### Step 2: Run the Go Webhook Receiver

To run the Go webhook receiver, navigate to the directory containing the `main.go` file and execute the following command:

```bash
go run main.go
```

The server will start and listen on port 5001.

### Step 3: Configure Alertmanager

In your Alertmanager configuration file (usually named `alertmanager.yml`), add a new receiver that points to the Go webhook receiver:

```yaml
route:
  receiver: 'webhook'
receivers:
- name: 'webhook'
  webhook_configs:
  - url: 'http://localhost:5001/'
```

### Step 4: Test the Setup

Trigger an alert in Prometheus to test the setup. If everything is configured correctly, the alert data should be printed by the Go webhook receiver.

**Comment by wooyoungAhn** [comment]
@rahulguptajss Thank you for your help.
oh That good idea we can use webhook!!

Thank you so much I will try to this.
